javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +520,356 @@ object Character {
   @inline def compare(x: scala.Char, y: scala.Char): Int =
     x - y
 
+  // UnicodeBlock
+
+  class Subset protected (name: String) {
+    final override def equals(that: Any): scala.Boolean = super.equals(that)
+    final override def hashCode: scala.Int = super.hashCode
+    final override def toString(): String = name
+  }
+
+  final class UnicodeBlock private (name: String, private val start: Int, private val end: Int) extends Subset(name)
+
+  object UnicodeBlock {
+    private[this] val allBlocks: ArrayList[UnicodeBlock] = new ArrayList[UnicodeBlock](220)
+    private[this] val blocksByNormalizedName = new java.util.HashMap[String, UnicodeBlock]()
+
+    private[this] def addNameAliases(properName: String, block: UnicodeBlock): Unit = {
+      // Add normalized aliases
+      val lower = properName.toLowerCase
+      //   lowercase + spaces
+      blocksByNormalizedName.put(lower, block)
+      //   lowercase + no spaces
+      blocksByNormalizedName.put(lower.replaceAll("\\s", ""), block)
+      //   tokens as _
+      blocksByNormalizedName.put(lower.replaceAll("[\\s\\-_]", "_"), block)
+    }
+
+    private[this] def addUnicodeBlock(name: String, properName: String, start: Int, end: Int): UnicodeBlock =  {
+      val block = new UnicodeBlock(name, start, end)
+      allBlocks.add(block)
+      addNameAliases(properName, block)
+      block
+    }
+
+    // Special constants
+    val SURROGATES_AREA = new UnicodeBlock("SURROGATES_AREA",0x0, 0x0)
+    blocksByNormalizedName.put("SURROGATES AREA".toLowerCase, SURROGATES_AREA)
+    blocksByNormalizedName.put("SURROGATES_AREA".toLowerCase, SURROGATES_AREA)
```

Comment:
Consider "constant-folding" the `toLowerCase` calls:
```suggestion
    blocksByNormalizedName.put("surrogates area", SURROGATES_AREA)
    blocksByNormalizedName.put("surrogates_area", SURROGATES_AREA)
```